c++/src/kj/debug.h

```diff
@@ -694,4 +713,18 @@ static constexpr DebugExpressionStart MAGIC_ASSERT;
 }  // namespace _ (private)
 }  // namespace kj
 
+KJ_ANNOTATE_SIGNAL_SAFE_STRINGIFY(
+    DebugComparison<Left KJ_COMMA Right>,
```

Comment:
I don't see the point in making this signal-safe. KJ_ASSERT still isn't signal-safe because it uses exceptions, which seems like a pretty fundamental thing we can't solve.

Reply:
Look for `EW_SIGNAL_SAFE_REQUIRE` in Workers. That's where it's used. We could move these to KJ if we also moved `writeLineToStderr` & some of the macro machinery (not for this PR obviously).

Reply:
Seems like a simpler choice would have been to just not use magic asserts in that macro?

Reply:
I thought about it, but making `DebugComparison` work was actually a tiny fraction of the overall work so I felt it's easier to keep it.